### PR TITLE
[T5A2][T14-B1] Tan Min Rong Samson

### PR DESCRIPTION
**Pull request by samsontmr**
Ready for review

### CODE REVIEW

File: src/seedu/addressbook/commands/Command.java [hunk 1]
```diff
@@ -38,10 +38,17 @@ public static String getMessageForPersonListShownSummary(List<? extends ReadOnly
...
     /**
      * Executes the command and returns the result.
+     * @throws Exception 
      */
-    public abstract CommandResult execute();
+    public abstract CommandResult execute() throws Exception;
 
     /**
+     * 
+     * @return true for command types that mutate the data.
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/Command.java]: try to be more specific on this header comment; the abstract class is a contract that we are providing to the client classes, hence the documentation should be precise. e.g., think what happens when you try to add a duplicate entry to address book. does it still mutate data?

File: src/seedu/addressbook/logic/Logic.java [hunk 2]
```diff
@@ -85,7 +85,9 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+        if (command.isMutating()) {
+            storage.save(addressBook);
```
Comment (okkhoy) [inline src/seedu/addressbook/logic/Logic.java]: may be you need to check how `LogicTest` works now? does it fail?

**Comment by okkhoy** [comment]
@samsontmr 
Some comments added. Please ack & close the PR after reading comments.

**Comment by samsontmr** [comment]
acknowledged, thanks!

**Pull request closed without merging**